azuread/data_users.go

```diff
@@ -62,54 +126,83 @@ func dataSourceUsersRead(d *schema.ResourceData, meta interface{}) error {
 	client := meta.(*ArmClient).usersClient
 	ctx := meta.(*ArmClient).StopContext
 
-	var users []graphrbac.User
+	var users []*graphrbac.User
 	expectedCount := 0
 
+	gracefulError := d.Get("graceful_errors").(bool)
 	if upns, ok := d.Get("user_principal_names").([]interface{}); ok && len(upns) > 0 {
 		expectedCount = len(upns)
 		for _, v := range upns {
-			resp, err := client.Get(ctx, v.(string))
+			u, err := client.Get(ctx, v.(string))
 			if err != nil {
-				return fmt.Errorf("Error making Read request on AzureAD User with ID %q: %+v", v.(string), err)
+				if !gracefulError {
```

Comment:
Can we also check the response was a 404, to make sure we aren't ignoring other errors?

e.g.
```hcl
if !gracefulError && ar.ResponseWasNotFound(u.Response) {
	return fmt.Errorf("...")
}
```